src/libDirectoryService/DirectoryService.cpp

```diff
@@ -334,6 +334,11 @@ bool DirectoryService::ProcessSetPrimary(const bytes& message,
             "Starting consensus on ds block");
   RunConsensusOnDSBlock();
 
+  {
```

Comment:
Is this code for fixing something?

Reply:
aah,  this branch forked out of fix/cleanup-powsolns which waiting to be merged.

Reply:
Anyways, need to rebased once https://github.com/Zilliqa/Zilliqa/pull/1154 is merged.